src/v/http/tests/request_builder_test.cc

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "http/request_builder.h"
+
+#include <gtest/gtest.h>
+
+namespace bh = boost::beast::http;
+
+TEST(request_builder, test_default_build_is_error) {
+    http::request_builder rb;
+    auto req = rb.build();
+    ASSERT_TRUE(!req.has_value());
+    ASSERT_TRUE(req.error() == rb.default_state);
+}
+
+TEST(request_builder, test_with_everything) {
```

Comment:
We need a test for URL query params that need URL encoding.

Reply:
added a test `test_url_percent_encoding`, there is also a new test for the http utils encoding method.